### ISSUE TITLE
Unreliable solver_describe_decision()

### ISSUE THREAD

**Issue by mluscon** [openSUSE/libsolv#127]
Hi,

I am getting strange results, if I try to obtain a reason of packages removed in a transaction.

e.g.:
If the SOLVER_CLEANDEPS flag is set and I tried to remove package A that requires B and C:
A = SOLVER_REASON_UNIT_RULE (correct)
B = SOLVER_REASON_WEAKDEP (incorrect)
C = SOLVER_REASON_WEAKDEP (incorrect)
Package A was installed by "install A" in the transaction comprising A, B, C. Therefore the reason of B and C should be SOLVER_REASON_CLEANDEPS_ERASE.

Is there anything else what I can do for the further debugging?

**Comment by mlschroe** [comment]
Yes, tell me what libsolv version you're using. You can also send me a solver testcase so I can debug this.

**Comment by mluscon** [comment]
The version is git master checkout and testcase:
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Req: B >= 1-0
#>=Req: C >= 1-0
#>=Pkg: B 1 1 noarch
#>=Pkg: C 1 1 noarch
repo available 0 testtags <inline>
system x86_64 * system
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
poolflags implicitobsoleteusescolors
job erase pkg A-1-1.x86_64 [cleandeps]
job userinstalled pkg A-1-1.x86_64
```

**Comment by mluscon** [comment]
Were you able to reproduce it?

**Comment by mlschroe** [comment]
Sorry, I had to work on other projects the last weeks. Next week looks promising, though. Thanks for the reminde!

**Comment by mlschroe** [comment]
Ok, that was rather easy to fix. Fixed with commit 48b46d3fc0384808cf214c3fbae366088b71402e
